### ISSUE TITLE
[Docs]: docs say that testing Chrome Extension content scripts isn't supported

### ISSUE THREAD

**Issue by lucasrmendonca** [puppeteer/puppeteer#10655]
### Feature description

The Chrome Extension docs say that testing popup scripts and content scripts is currently not supported, but that's not true since Chrome for Testing and headless: 'new' were announced.

Maybe we should update them to say that support is experimental instead of unsupported?

If this makes sense, I'd be glad to make a PR

**Comment by OrKoN** [comment]
We would need to add tests demonstrating that it works in the new headless mode. Do you want to try to do that? It might be that Chrome for Testing and headless: new supports it but some pieces are still missing in Puppeteer.

**Comment by Josmithr** [comment]
My team is actively trying to get extension tests up and running. Can someone point us to any docs on how to do this now that it is supported?

**Comment by OrKoN** [comment]
@Josmithr see https://pptr.dev/guides/chrome-extensions (background_page  should be supported). service workers from extensions are probably supported-

**Comment by Josmithr** [comment]
@OrKoN Thank you! We're currently following that pattern (though we're using manifest v3). The page notes that content scripts are not yet supported - is there any further elaboration on that detail somewhere? We're trying to get automated tests going for some end-to-end scenarios, and we're currently trying to diagnose issues where messages aren't flowing as we expect. We're wondering if this may have to do with content script support limitations.

(Let me know if you would prefer this discussion move elsewhere - but this seems at least tangentially related to the completeness of the documentation here)

**Comment by OrKoN** [comment]
@Josmithr I don't have many details on this. The new headless mode supports extensions (same as the headful mode). So you can run extensions and they should work the same way they work in the regular browser. The limited support here is whether you can connect to content scripts via Chrome DevTools Protocol/Puppeteer. In general, you should be able to connect to the background page using CDP, trigger some actions and observe the changes made by the content scripts to the target page. I don't think it's possible to intercept the content scripts in some way or evaluate code in the content script yet.

> The Chrome Extension docs say that testing popup scripts and content scripts is currently not supported, but that's not true since Chrome for Testing and headless: 'new' were announced.

To clarify, although the new headless fully supports extensions it does not mean that the CDP support is there for all extension features. 

> messages aren't flowing as we expect between our content and background scripts. We're wondering if this may have to do with content script support limitations.

It is very likely not related.

**Comment by nroscino** [comment]
Hey @OrKoN!
Puppeteer docs was recently updated with the newly introduced Realms API. This allows interaction with content scripts. Do you think we can close this?

**Comment by OrKoN** [comment]
Yes, once https://pptr.dev/guides/chrome-extensions#content-scripts is updated. Please update the info and add an example on how to access extension realms.

**Comment by nroscino** [comment]
The doc was updated here: https://github.com/puppeteer/puppeteer/pull/14924.
